doc/source/serve/autoscaling.md

```diff
@@ -0,0 +1,43 @@
+(serve-autoscaling)=
+
+# Serve Autoscaling
+
+This section should help you:
+
+- Understand how Ray Serve autoscaling works.
+- Learn how to configure the parameters for your workload.
+
+
+## Autoscaling architecture
+Ray Serve's autoscaling feature automatically increases or decreases a deployment's number of replicas based on its load. 
+
+![pic](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/autoscaling.svg)
+
+- Each ServeHandle and each worker replica periodically push the stats to the autoscaler.
+- The autoscaler requires ServeHandle queue metrics and replicas queries metrics to make decision whether to scale (up or down) the replicas.
+- ServeHandle continues to poll the updated group of replicas from the controller. Upon discovery of the new replicas, it will send any buffered or new queries to the replica until `max_concurrent_queries` is reached
+
+:::{note}
+When the controller dies, the client will still be able to send queries, but autoscaling will be paused. When the controller recovers, the autoscaling will resume, but all previous metrics collected will be lost.
+:::
+
+## Autoscaling parameters
+There are several parameters the autoscaling algorithm takes into consideration when deciding the target replicas for your deployment
+**min_replicas**: The minimal number of replicas for the deployment. ``min_replicas`` will also be the initial number of replicas when the deployment is deployed.
+:::{note}
+Ray Serve Autoscaling allows the `min_replicas` to be 0 when starting your deployment; the scale up will be started when you start sending traffic. There will be a cold start time as the Ray ServeHandle waits (blocks) for available replicas to assign the request.
+:::
+**max_replicas**: Max replicas is the maximum number of replicas for the deployment. Ray Serve Autoscaling will rely on the Ray Autoscaler to scale up more nodes when the currently available cluster resources (CPUs, GPUs, etc) are not enough to bring up more replicas. 
+**target_num_ongoing_requests_per_replica**: The config is to maintain how many ongoing requests are expected to run concurrently per replica at most. If the number is lower, the scale up will be done more aggressively.
```

Comment:
Nice change! Since this value isn't a hard limit, could we mention that at the end of the description? We could add a sentence saying "Note that this value isn't a hard limit. Under x, y, z circumstances, there could be more ongoing requests per replica than this value."